src/Composer/ExtensionInstaller.php

```diff
@@ -12,13 +12,19 @@ class ExtensionInstaller
      *
      * @param \Composer\EventDispatcher\Event $event
      */
-    public static function handle(Event $event)
+    public static function handle($event)
     {
-        if (!($event instanceof ScriptEvent || $event instanceof PackageEvent)) {
+        try {
+            $operation = $event->getOperation();
+            if (method_exists($operation, 'getPackage')) {
+                $installedPackage = $operation->getPackage();
+            } elseif (method_exists($operation, 'getTargetPackage')) {
+                $installedPackage = $operation->getTargetPackage();
+            }
```

Comment:
Shouldn't we also:
```php
} else {
    return;
}
```
… to prevent anything else randomly b0rking, if we've missed something, or Composer changes.